src/Geometry/Surface.jl

```diff
@@ -27,6 +27,7 @@ Basic `Surface`s are _not_ valid CSG objects, they function only in a stand-alon
 ```julia
 surfaceintersection(surface::Surface{T}, ray::AbstractRay{T,3}) -> Union{EmptyInterval{T},Interval{T}}
 normal(surface::Surface{T}) -> SVector{3,T}
+surface_id(surface::Surface{T}) -> UUID
```

Comment:
We could define interface and surface_id once in Surface.jl like this:

surface_id(surf::Surface) = surf.surface_id
interface(surf::Surface) = surf.interface

Then all surfaces would inherit these functions and we wouldn't have to replicate them. A few surfaces need special surface_id and interface functions such as Rectangle

surface_id(surf::Surface) = surface_id(surf.plane)

but that should work fine. If a function defines a more specialized surface_id or interface function that's the one that will be executed rather than the default defined in Surface.jl.

Not essential to make this change but would simplify the code a bit.